udata_hydra/routes/status.py

```diff
@@ -30,41 +27,84 @@ async def get_status_counts(request: web.Request) -> dict[str, int]:
+    # - Don't have at least two checks yet, and the last check is older than CHECK_DELAYS[0]
+    # - At least one the their two most recent last checks have no detected_last_modified_at, and the last check is older than CHECK_DELAYS[0]
+    # - Their two most recent last checks have changed, and the last check is older than CHECK_DELAYS[0]
+    # - Their last two checks have not changed, the two checks have done between two delays in CHECK_DELAYS, and the last check is older than the same delay in CHECK_DELAYS (this is in order to avoid checking too often the same resource which doesn't change)
+    q_start = f"""
+        WITH recent_checks AS (
+            SELECT resource_id, detected_last_modified_at, created_at,
+                ROW_NUMBER() OVER (PARTITION BY resource_id ORDER BY created_at DESC) AS rn
+            FROM checks
+        )
+        SELECT COUNT(*) AS count_outdated
+        FROM (
+            SELECT catalog.url, dataset_id, catalog.resource_id
+            FROM catalog
+                LEFT JOIN recent_checks ON catalog.resource_id = recent_checks.resource_id
+            WHERE catalog.priority = False
+                AND {Resource.get_excluded_clause()}
+            GROUP BY catalog.url, dataset_id, catalog.resource_id
+            HAVING (
+                (
+                    COUNT(recent_checks.resource_id) < 2
+                    AND MAX(recent_checks.created_at) < NOW() AT TIME ZONE 'UTC' - INTERVAL '{config.CHECK_DELAYS[0]}'
+                )
+                OR (
+                    COUNT(CASE WHEN recent_checks.detected_last_modified_at IS NULL THEN 1 END) >= 1
+                    AND MAX(recent_checks.created_at) < NOW() AT TIME ZONE 'UTC' - INTERVAL '{config.CHECK_DELAYS[0]}'
+                )
+                OR (
+                    COUNT(DISTINCT recent_checks.detected_last_modified_at) > 1
+                    AND MAX(recent_checks.created_at) < NOW() AT TIME ZONE 'UTC' - INTERVAL '{config.CHECK_DELAYS[0]}'
+                )
+    """
+    q_dynamic = ""
+    for i in range(len(config.CHECK_DELAYS) - 1):
+        q_dynamic += f"""
+                OR (
+                    COUNT(recent_checks.resource_id) = 2
+                    AND COUNT(DISTINCT recent_checks.detected_last_modified_at) < 2
+                    AND MAX(recent_checks.created_at) < NOW() AT TIME ZONE 'UTC' - INTERVAL '{config.CHECK_DELAYS[i]}'
+                    AND MAX(recent_checks.created_at) >= NOW() AT TIME ZONE 'UTC' - INTERVAL '{config.CHECK_DELAYS[i + 1]}'
+                    AND MIN(recent_checks.created_at) < MAX(recent_checks.created_at) - INTERVAL '{config.CHECK_DELAYS[i]}'
+                    AND MIN(recent_checks.created_at) >= MAX(recent_checks.created_at) - INTERVAL '{config.CHECK_DELAYS[i + 1]}'
+                )
+        """
+
+    q_end = f"""
+                OR MAX(recent_checks.created_at) < NOW() AT TIME ZONE 'UTC' - INTERVAL '{config.CHECK_DELAYS[-1]}'
+            )
+        ) AS subquery;
```

Comment:
I'm afraid I'd be unable to validate that this works properly without the tests to back it up, and I'm not even sure I understand the various cases explained in the comment above the query, I'm sorry :(

Reply:
Damned, I completely forgot about this file, that's actually a mistake! It's an old commit that shouldn't be there. The idea is to avoid this previous way of doing it with a overly complicated and unmaintainable SQL query.
Will remove a write a bit

Reply:
Thanks for noticing!

Reply:
Fixed!